packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx

```diff
@@ -43,20 +316,56 @@ const baseRenderLeaf = (props: RenderLeafProps, modifiers: ModifiersStore) => {
   return <span {...props.attributes}>{wrappedChildren}</span>;
 };
 
-const baseRenderElement = (props: RenderElementProps, blocks: BlocksStore) => {
+type BaseRenderElementProps = Direction & {
+  props: RenderElementProps['children'];
+  blocks: BlocksStore;
+  editor: Editor;
+};
+
+const isLink = (element: Element): element is Block<'link'> => {
+  return element.type === 'link';
+};
+
+const isList = (element: Element): element is Block<'list'> => {
+  return element.type === 'list';
+};
```

Comment:
Exporting these type guards is not allowed, I get below error for already defined types under attributes/blocks.

`Exported variable 'isLink' has or is using name 'LinkInlineNode' from external module "/Users/madhuri/Dev/strapi/packages/core/types/dist/types/core/attributes/blocks" but cannot be named.`

Reply:
You probably need to export some of the types in that case then, I wouldn't duplicate code because of this error, i would solve the issue because you won't be the last person to find it haha